sequencer/src/api/sql.rs

```diff
@@ -112,10 +125,67 @@ impl CatchupDataSource for DataSource {
     }
 }
 
+#[async_trait]
+impl ChainConfigPersistence for SqlStorage {
+    async fn insert_chain_config(&mut self, chain_config: ChainConfig) -> anyhow::Result<()> {
+        let commitment = chain_config.commitment();
+        let data = bincode::serialize(&chain_config)?;
+
+        let mut transaction = self.transaction().await?;
+
+        transaction
+            .upsert(
+                "chain_config",
+                ["commitment", "data"],
+                ["commitment"],
+                [[sql_param(&(commitment.to_string())), sql_param(&data)]],
+            )
+            .await?;
+
+        self.commit().await?;
```

Comment:
we do revert in the update_state_storage fn where this is called

Reply:
Hm. Then I think we should not commit here. I think we can have functions that do entire transactions themselves, and functions that don't commit/revert and are meant to be called as part of a larger transaction. But having the commit so far separated from the revert like this makes me nervous

Reply:
I really want to make an RAII guard version of `transaction` in the query service. That is almost always what you want anyways. The query service update API is sort of a special case where `insert_block`/`insert_leaf` are called and implicitly operate in the current transaction. But for everything else, having a guard is better. And for functions like this, we could have it take in the guard by `&mut`, which enforces at the type system that this is part of atransaction and does not finalize the transaction (since `commit`/`revert` would consume the guard).

Haven't had time to get around to that though :(

Reply:
https://github.com/EspressoSystems/hotshot-query-service/issues/634